pkg/kclient/ingress.go

```diff
@@ -0,0 +1,167 @@
+	ingress := &extensionsv1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   ingressParam.Name,
+			Labels: labels,
+		},
+		Spec: extensionsv1.IngressSpec{
+			Rules: []extensionsv1.IngressRule{
+				{
+					Host: ingressParam.IngressDomain,
+					IngressRuleValue: extensionsv1.IngressRuleValue{
+						HTTP: &extensionsv1.HTTPIngressRuleValue{
+							Paths: []extensionsv1.HTTPIngressPath{
+								{
+									Path: "/",
+									Backend: extensionsv1.IngressBackend{
+										ServiceName: ingressParam.ServiceName,
+										ServicePort: ingressParam.PortNumber,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	secretNameLength := len(ingressParam.TLSSecretName)
+	if secretNameLength != 0 {
+		ingress.Spec.TLS = []extensionsv1.IngressTLS{
+			{
+				Hosts: []string{
+					ingressParam.IngressDomain,
+				},
+				SecretName: ingressParam.TLSSecretName,
+			},
+		}
+	}
+
+	r, err := c.KubeClient.ExtensionsV1beta1().Ingresses(c.Namespace).Create(ingress)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating ingress")
+	}
+	return r, nil
+}
+
+// DeleteIngress deleted the given route
+func (c *Client) DeleteIngress(name string) error {
+	ingress, err := c.KubeClient.ExtensionsV1beta1().Ingresses(c.Namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, "unable to get ingress")
+	}
+	ingressTLSArray := ingress.Spec.TLS
+	for _, elem := range ingressTLSArray {
+		err = c.KubeClient.CoreV1().Secrets(c.Namespace).Delete(elem.SecretName, &metav1.DeleteOptions{})
+		if err != nil {
+			return errors.Wrap(err, "unable to delete tls secret")
+		}
+	}
+
+	err = c.KubeClient.ExtensionsV1beta1().Ingresses(c.Namespace).Delete(name, &metav1.DeleteOptions{})
+	if err != nil {
+		return errors.Wrap(err, "unable to delete ingress")
+	}
+	return nil
+}
+
+// ListIngresses lists all the ingresses based on the given label selector
+func (c *Client) ListIngresses(labelSelector string) ([]extensionsv1.Ingress, error) {
+	ingressList, err := c.KubeClient.ExtensionsV1beta1().Ingresses(c.Namespace).List(metav1.ListOptions{
+		LabelSelector: labelSelector,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get ingress list")
+	}
+
+	return ingressList.Items, nil
+}
+
+// ListIngressNames lists all the names of the ingresses based on the given label
+// selector
+func (c *Client) ListIngressNames(labelSelector string) ([]string, error) {
+	ingresses, err := c.ListIngresses(labelSelector)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to list ingresses")
+	}
+
+	var ingressNames []string
+	for _, i := range ingresses {
+		ingressNames = append(ingressNames, i.Name)
+	}
+
+	return ingressNames, nil
+}
+
+// ListSecrets lists all the secrets based on the given label selector
+func (c *Client) ListSecrets(labelSelector string) ([]corev1.Secret, error) {
```

Comment:
This seems like a common function for secrets. I think we should create a `secrets.go` file in the kclient package and move it there.

Reply:
updated